app/src/main/java/com/brewthings/app/util/Calculations.kt

```diff
@@ -16,12 +16,9 @@ fun calculateVelocity(previousData: RaptPillData?, fgData: RaptPillData): Float?
     val gpDrop = (fgData.gravity - previousData.gravity) * 1000f
     val daysBetween = (fgData.timestamp.epochSeconds - previousData.timestamp.epochSeconds).toFloat() / 86_400f
     val velocity = gpDrop / daysBetween
-    return velocity.sanitizeVelocity()
+    if (velocity.validateVelocity()) return velocity else return null
 }
 
-fun Float.sanitizeVelocity(): Float? =
-    if (isInfinite() || isNaN() || this > 0 || this < -100) {
-        null // Invalid velocity.
-    } else {
-        -1 * this // Invert the sign, to make it more intuitive.
-    }
+fun Float.validateVelocity(): Boolean = (isInfinite() || isNaN() || this > 0 || this < -100)
```

Comment:
No need for the parhentesis
```suggestion
fun Float.validateVelocity(): Boolean = isInfinite() || isNaN() || this > 0 || this < -100
```

Reply:
I have another suggestion based on the usage of this function. why not doing
```suggestion
fun Float.takeIfIsValidVelocity(): Float? = takeIf { it.isInfinite() || it.isNaN() || it > 0 || it < -100 }
```

so that, in `Calculations.kt` it becomes
```kotlin
return velocity.takeIfIsValidVelocity()
```

and in `RaptPillParser.kt`
```kotlin
rawVelocity = if (gravityVelocityValid) gravityVelocity.takeIfIsValidVelocity(),
```

Reply:
I think there is also a big mistake: in the original code, the condition was checking for an **invalid** velocity
```kotlin
if (isInfinite() || isNaN() || this > 0 || this < -100) {
        null // Invalid velocity.
    }
```

so, validateVelocity will now return `true` when it's `not` valid
```kotlin
fun Float.validateVelocity(): Boolean = (isInfinite() || isNaN() || this > 0 || this < -100)
```

so, this won't work
```kotlin
if (velocity.validateVelocity()) return velocity else return null
```